src/schema/tye-schema.json

```diff
@@ -383,6 +383,13 @@
                     "description": "The Dockerfile to use.",
                     "type": "string"
                 },
+                "dockerFileArgs": {
+                    "description": "Build arguments to use when building the image.",
+                    "type": "array",
+                    "items": {
+                      "type": "object"
```

Comment:
Should the `items` type be `string`?

Reply:
Or, rather, whether there's a better way to model a string --> string dictionary in JSON schema.

Reply:
Looks like that would give a syntax error, from what I saw it really is an object

![image](https://user-images.githubusercontent.com/23037278/123468736-9252af80-d5c8-11eb-8678-e0912d156e5f.png)

https://github.com/dotnet/tye/blob/5096dc54c3dc9daf71f67b2ebd919f726ca3aaf6/test/E2ETest/testassets/projects/single-phase-dockerfile-multi-args/tye.yaml#L7-L14

Reply:
Yes, it looks like we accept whatever the string version of the YAML scalar node, so it's essentially a string --> object.  In that case, I think that change fine.